backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -393,22 +409,41 @@ const testUploadAvatar = () => {
 		});
 
 		test('should fail if the user has a session via an API key', async () => {
-			const res = await agent.put(`/v5/user/avatar?key=${testUser.apiKey}`)
+			const res = await agent.put(`/v5/user/avatar?key=${userWithGridFsAvatar.apiKey}`)
 				.expect(templates.notLoggedIn.status);
 			expect(res.body.code).toEqual(templates.notLoggedIn.code);
 		});
 
+		describe('With valid authentication and existing avatar', () => {
+			beforeAll(async () => {
+				await testSession.post('/v5/login/').send({ user: userWithGridFsAvatar.user, password: userWithGridFsAvatar.password });
+			});
+			afterAll(async () => {
+				await testSession.post('/v5/logout/');
+			});
+
+			test('should remove old avatar and upload a new one if the user is logged in', async () => {
+				await testSession.put('/v5/user/avatar').set('Content-Type', 'image/png').attach('file', image)
+					.expect(templates.ok.status);
+				// ensure an avatar exists
+				await testSession.get('/v5/user/avatar').expect(200);
```

Comment:
should be checking the image is the new one

Reply:
the problem is that it returns a buffer... should I try to convert the image Im passing to a buffer and compare them ?

Reply:
'image' is the path to a file

Reply:
sure, but you can read the file and check it against the response?